File: src/main/java/org/dukcode/ps/codetree/trail02/chapter02/lesson01/IntroToOutputNumerically.java
package org.dukcode.ps.codetree.trail02.chapter02.lesson01;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.StringTokenizer;

/**
 * @see <a
 * href="https://www.codetree.ai/ko/trails/complete/curated-cards/intro-to-output-numerically/description">LINK</a>
 */
public class IntroToOutputNumerically {

  private static BufferedReader br;
  private static BufferedWriter bw;
  private static StringTokenizer st;

  public static void main(String[] args) throws IOException {
    br = new BufferedReader(new InputStreamReader(System.in));
    bw = new BufferedWriter(new OutputStreamWriter(System.out));

    int n = Integer.parseInt(br.readLine());

    printIncrease(n);
    bw.newLine();
    printDecrease(n);
    bw.newLine();

    br.close();
    bw.close();

  }

  private static void printIncrease(int n) throws IOException {
    if (n == 0) {
      return;
    }

    printIncrease(n - 1);

    bw.write(String.valueOf(n));
    bw.write(' ');
  }

  private static void printDecrease(int n) throws IOException {
    if (n == 0) {
      return;
    }

    bw.write(String.valueOf(n));
    bw.write(' ');

    printDecrease(n - 1);
  }
}

File: src/main/kotlin/org/dukcode/ps/codetree/trail02/chapter01/lesson05/IntroToOutputNumerically.kt
package org.dukcode.ps.codetree.trail02.chapter01.lesson05.introToOutputNumerically

import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter

private val br = BufferedReader(InputStreamReader(System.`in`))
private val bw = BufferedWriter(OutputStreamWriter(System.out))

/**
 * @see <a href="https://www.codetree.ai/ko/trails/complete/curated-cards/intro-to-output-numerically/description">LINK</a>
 */
fun main() {
    val n = br.readLine().toInt()

    printAscending(n)
    bw.newLine()

    printDescending(n)
    bw.newLine()

    bw.flush()
}

fun printDescending(n: Int) {
    if (n == 0) {
        return
    }

    bw.write(n.toString())
    bw.write(' '.code)

    printDescending(n - 1)
}

fun printAscending(n: Int) {
    if (n == 0) {
        return
    }

    printAscending(n - 1)

    bw.write(n.toString())
    bw.write(' '.code)
}
